src/pymodaq/utils/parameter/pymodaq_ptypes/itemselect.py

```diff
@@ -68,7 +68,6 @@ def set_value(self, values):
         if allitems != values['all_items']:
             self.clear()
             for value in values['all_items']:
-                # if value not in allitems:
                 item = QtWidgets.QListWidgetItem(value)
                 item.setFlags(item.flags() | QtCore.Qt.ItemIsUserCheckable)
```

Comment:
is it not missing here the option checking to see if one should add the usercheckable item?
something like:
```
if 'check' in opts and opts['check'] is True:
    item.setFlags(item.flags() | QtCore.Qt.ItemIsUserCheckable)
else:
    item.setFlags(item.flags())
```

Reply:
I do not think that it is missing. The way I did it was by adding a new class ItemCheck which is very similar to ItemSelect. 
I chose this option to immediately differ from the start if the checkbox should be used or just standard item selection.  The reason for it is that they use different functions overall:
- the signal connection: item.changed vs item.SelectionChanged
- the selection: self.selectedItems vs self.all_items() if item.checkState()!=0
- the initialization: self.addItem vs item, item.setFlag, item.setCheckState, self.addItem

However, I am ok to use a single class giving the opts checkbox in the init if you prefer.